packages/lodestar-beacon-state-transition/src/util/blockRoot.ts

```diff
@@ -25,20 +27,27 @@ export function getBlockRootAtSlot(config: IBeaconConfig, state: phase0.BeaconSt
 /**
  * Return the block root at the start of a recent [[epoch]].
  */
-export function getBlockRoot(config: IBeaconConfig, state: phase0.BeaconState, epoch: Epoch): Root {
+export function getBlockRoot(config: IBeaconConfig, state: BeaconState, epoch: Epoch): Root {
   return getBlockRootAtSlot(config, state, computeStartSlotAtEpoch(config, epoch));
 }
 /**
  * Return the block header corresponding to a block with ``state_root`` set to ``ZERO_HASH``.
  */
-export function getTemporaryBlockHeader(config: IBeaconConfig, block: phase0.BeaconBlock): phase0.BeaconBlockHeader {
+export function getTemporaryBlockHeader(
+  config: IBeaconConfig,
+  block: phase0.BeaconBlock | lightclient.BeaconBlock,
+  type: ContainerType<phase0.BeaconBlockBody> | ContainerType<lightclient.BeaconBlockBody> | null = null
+): phase0.BeaconBlockHeader {
+  if (!type) {
+    type = config.types.phase0.BeaconBlockBody;
+  }
   return {
     slot: block.slot,
     proposerIndex: block.proposerIndex,
     parentRoot: block.parentRoot,
     // `state_root` is zeroed and overwritten in the next `process_slot` call
     stateRoot: ZERO_HASH,
-    bodyRoot: config.types.phase0.BeaconBlockBody.hashTreeRoot(block.body),
+    bodyRoot: type.hashTreeRoot(block.body as lightclient.BeaconBlockBody),
```

Comment:
Should be able to do this without passing in a type with: `config.getTypes(block.slot).BeaconBlockBody.hashTreeRoot(block.body)`

Reply:
You would think that but spec test blocks don't have slot after hf1 fork slot

Reply:
we just need to create new `IBeaconConfig`s for the spec tests.
```ts
import {params as minimalParams} from "@chainsafe/lodestar-params/minimal";
import {params as mainnetParams} from "@chainsafe/lodestar-params/mainnet";

export const lightclientSpecTestMinimal = createIBeaconConfig({...minimalParams, LIGHTCLIENT_PATCH_FORK_SLOT: 0});
export const lightclientSpecTestMainnet = createIBeaconConfig({...mainnetParams, LIGHTCLIENT_PATCH_FORK_SLOT: 0});
```